### PR TITLE
[BEAM-5790] Euphoria: Remove Dataset abstraction.

### PR DESCRIPTION
**Pull request by dmvk**
This removes a dataset abstraction, which is not needed anymore. It should provide more seamless integration with Beam Java SDK.

### CODE REVIEW

**Comment by dmvk** [comment]
@mareksimunek @VaclavPlajt does any of you have any concerns about this?

**Comment by dmvk** [comment]
@VaclavPlajt it is updated already.

File: sdks/java/extensions/euphoria/src/main/java/org/apache/beam/sdk/extensions/euphoria/core/client/operator/FlatMap.java [hunk 1]
```diff
@@ -160,11 +160,7 @@ public static OfBuilder named(@Nullable String name) {
    * Last builder in a chain. It concludes this operators creation by calling {@link
    * #output(OutputHint...)}.
    */
-  public interface OutputBuilder<OutputT> extends Builders.Output<OutputT> {
-
-    @Override
-    Dataset<OutputT> output(OutputHint... outputHints);
-  }
+  public interface OutputBuilder<OutputT> extends Builders.Output<OutputT> {}
```
Comment (mxm) [inline sdks/java/extensions/euphoria/src/main/java/org/apache/beam/sdk/extensions/euphoria/core/client/operator/FlatMap.java]: Is this interface still necessary?

File: sdks/java/extensions/euphoria/src/test/java/org/apache/beam/sdk/extensions/euphoria/core/translate/EuphoriaTest.java [hunk 2]
```diff
@@ -1,103 +0,0 @@
...
-import org.apache.beam.sdk.testing.TestPipeline;
-import org.apache.beam.sdk.transforms.Create;
-import org.apache.beam.sdk.values.KV;
-import org.apache.beam.sdk.values.PCollection;
-import org.junit.Before;
-import org.junit.Rule;
-import org.junit.Test;
-
-/** A group of test focused at {@link Euphoria}. */
-public class EuphoriaTest implements Serializable {
```
Comment (mxm) [inline sdks/java/extensions/euphoria/src/test/java/org/apache/beam/sdk/extensions/euphoria/core/translate/EuphoriaTest.java]: I suppose we have more specific tests and don't need this one anymore?

File: website/src/documentation/sdks/euphoria.md [hunk 3]
```diff
@@ -104,46 +97,31 @@ pipeline.run();
 
 Euphoria API is composed from a set of operators, which allows you to construct `Pipeline` according to your application needs.
 
-### Datasets
-Euphoria uses the concept of 'Datasets' to describe data pipeline between `Operators`. This concept is similar to Beam's `PCollection` and can be converted back and forth through:
```
Comment (mxm) [inline website/src/documentation/sdks/euphoria.md]: Should we mention Beam's `PCollection` here or do we just assume users are familiar with it?

File: sdks/java/extensions/euphoria/src/test/java/org/apache/beam/sdk/extensions/euphoria/core/translate/EuphoriaTest.java (hunk 2 above)
Comment (dmvk) [reply sdks/java/extensions/euphoria/src/test/java/org/apache/beam/sdk/extensions/euphoria/core/translate/EuphoriaTest.java]: This test was for the Euphoria PTransform -> `PTransform<Dataset<In>, Dataset<Out>>`, that was removed by this PR.

File: sdks/java/extensions/euphoria/src/main/java/org/apache/beam/sdk/extensions/euphoria/core/client/operator/FlatMap.java (hunk 1 above)
Comment (dmvk) [reply sdks/java/extensions/euphoria/src/main/java/org/apache/beam/sdk/extensions/euphoria/core/client/operator/FlatMap.java]: It is not 👍 I've removed the extra interface where possible.

File: website/src/documentation/sdks/euphoria.md (hunk 3 above)
Comment (dmvk) [reply website/src/documentation/sdks/euphoria.md]: Right now we are assuming that user is familiar with the basic beam model, which has great documentation. We may reconsider this, if we see new users struggling.

**Pull request merged**